front/app/components/SignUpIn/SignUp/PasswordSignup.tsx

```diff
@@ -388,23 +399,27 @@ class PasswordSignup extends PureComponent<Props & InjectedIntlProps, State> {
         intl: { formatMessage },
         locale,
       } = this.props;
+
       const {
         token,
         firstName,
         lastName,
         emailOrPhoneNumber,
         password,
         processing,
+      } = this.state;
+
+      // get these directly from the validate function
+      // rather than pulling from state, to make sure we have updated data
+      const {
         invitationRedeemError,
         emailOrPhoneNumberError,
         firstNameError,
         lastNameError,
         hasMinimumLengthError,
         tacError,
         privacyError,
-      } = this.state;
-
-      this.validate(isPhoneSignupEnabled);
+      } = this.validate(isPhoneSignupEnabled);
 
       const hasErrors = [
```

Comment:
I wonder if it wouldn't be easier to just add all this logic of checking for errors inside `validate` and only return hasErrors as a boolean. Then you can just do `const hasErrors = this.validate(isPhoneSignupEnabled);` and go from there?